week-3/mandatory/2-api/server.js

```diff
@@ -0,0 +1,307 @@
+        res.status(200).json({ customers: result.rows });
+      } else {
+        console.log(err);
+        res.status(400).json(err);
+      }
+    }
+  );
+});
+
+//  Get Products & filtered Products on the same path---
+
+// ---- Solution 1------------
+
+app.get("/products", function (req, res) {
+  const name = req.query.name;
+
+  let allProducts = `SELECT products.product_name, pa.unit_price price, sup.supplier_name
+  FROM products JOIN product_availability pa ON (products.id=pa.prod_id)
+  JOIN suppliers sup ON (pa.supp_id=sup.id) `;
+
+  let filteredProducts =
+    allProducts + " WHERE LOWER(products.product_name) LIKE '%'||$1||'%'";
+
+  !name
+    ? db.query(allProducts, [], (error, result) => {
+        if (!error) res.json(result.rows);
+      })
+    : db.query(filteredProducts, [name.toLowerCase()], (error, result) => {
+        if (!error) res.json(result.rows);
+      });
+});
+
+// ---- Solution 2------------
+
+// app.get("/products", function (req, res) {
+//   const name = req.query.name;
+
+//   let query = `SELECT products.product_name, pa.unit_price price, sup.supplier_name
+//   FROM products JOIN product_availability pa ON (products.id=pa.prod_id)
+//   JOIN suppliers sup ON (pa.supp_id=sup.id)`;
+
+//   let param = [];
+
+//   if (name !== undefined) {
+//     query = `SELECT products.product_name, pa.unit_price price, sup.supplier_name
+//   FROM products JOIN product_availability pa ON (products.id=pa.prod_id)
+//   JOIN suppliers sup ON (pa.supp_id=sup.id) WHERE LOWER(products.product_name) LIKE '%'||$1||'%'`;
+
+//     param = [name.toLowerCase()];
+//   }
+
+//   db.query(query, param)
+//     .then((result) => res.json(result.rows))
+//     .catch((e) => console.log(e));
+// });
+
+// ------ Create New Cust -----------
+
+app.post("/customers", (req, res) => {
+  const custName = req.body.name;
+  const custAddress = req.body.address;
+  const custCity = req.body.city;
+  const custCountry = req.body.country;
+  db.query(
+    "INSERT INTO customers (name, address, city, country) " +
+      "VALUES ($1, $2, $3, $4) " +
+      "RETURNING id",
+    [custName, custAddress, custCity, custCountry],
+    (err) => {
+      if (err === undefined) {
+        res.send("New customer added");
+      } else {
+        console.log(err);
+        res.status(500).json({ error: err });
+      }
+    }
+  );
+});
+
+// --------- Create New Product ---------
+
+app.post("/products", (req, res) => {
+  const productName = req.body.product_name;
+
+  db.query(
+    "INSERT INTO products (product_name) VALUES ($1) RETURNING id",
+    [productName],
+    (err) => {
+      if (!err) res.send("New Producted Added");
+      res.status(500).json({ error: err });
+      console.log(err);
+    }
+  );
+});
+
+// !! ----------Create New Availability -----
+
+app.post("/availability", (req, res) => {
+  const suppId = req.body.supp_id;
+  const price = req.body.unit_price;
+
+  if (price <= 0 || !Number.isInteger(price)) {
```

Comment:
price is a string, so both conditions will fail, as you need to cast it to number first